server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -160,7 +183,10 @@ void doRun() {
             // Stop cancelling tasks if there are no tokens in either of the two token buckets.
             if (rateLimitReached && ratioLimitReached) {
                 logger.debug("task cancellation limit reached");
-                state.incrementLimitReachedCount();
+                SearchBackpressureState searchBackpressureState = searchBackpressureStates.get(
+                    (taskCancellation.getTask() instanceof SearchTask) ? SearchTask.class : SearchShardTask.class
+                );
+                searchBackpressureState.incrementLimitReachedCount();
```

Comment:
Extension to my previous comment: The cancellation rate limiter is shared among all task types but the counters are not. This may be fine but needs to be clearly conveyed through documentation.

Reply:
I have separated the rate and ratio limiters for both task types.